src/DebugConfigurationProvider.ts

```diff
@@ -0,0 +1,80 @@
+import * as vscode from 'vscode'
+import { join } from 'path'
+import { readFileSync } from 'fs'
+
+function tryGetCRACommand(rootPath: string): string {
+  // Known binary names of `react-scripts` forks:
+  const packageBinaryNames = ['react-scripts', 'react-native-scripts', 'react-scripts-ts', 'react-app-rewired']
+  // If possible, try to parse `package.json` and look for a known binary beeing called in `scripts.test`
+  try {
+    const packagePath = join(rootPath, 'package.json')
+    const packageJSON = JSON.parse(readFileSync(packagePath, 'utf8'))
+    if (!packageJSON || !packageJSON.scripts || !packageJSON.scripts.test) {
+      return ''
+    }
+    const testCommand = packageJSON.scripts.test as string
+    if (packageBinaryNames.some(binary => testCommand.indexOf(binary + ' test') === 0)) {
+      return testCommand
+    }
+  } catch {}
+  return ''
+}
+
+export class NodeDebugConfigurationProvider implements vscode.DebugConfigurationProvider {
+  private fileNameToRun: string = ''
+  private testToRun: string = ''
+
+  public prepareTestRun(fileNameToRun: string, testToRun: string) {
+    this.fileNameToRun = fileNameToRun
+    this.testToRun = testToRun
+  }
+
+  resolveDebugConfiguration(
+    _folder: vscode.WorkspaceFolder | undefined,
+    debugConfiguration: vscode.DebugConfiguration,
+    _token?: vscode.CancellationToken
+  ) {
+    if (debugConfiguration.name !== 'vscode-jest-tests') {
+      return debugConfiguration
+    }
+    // necessary for running CRA test scripts in non-watch mode
+    if (debugConfiguration.env) {
+      debugConfiguration.env.CI = 'vscode-jest-tests'
+    } else {
+      debugConfiguration.env = { CI: 'vscode-jest-tests' }
+    }
+    if (this.fileNameToRun) {
+      debugConfiguration.args.push(this.fileNameToRun)
+      if (this.testToRun) {
+        debugConfiguration.args.push('--testNamePattern')
+        debugConfiguration.args.push(this.testToRun)
+      }
+      this.fileNameToRun = ''
+      this.testToRun = ''
+    }
+    return debugConfiguration
+  }
+}
+
+export class SnippetDebugConfigurationProvider implements vscode.DebugConfigurationProvider {
```

Comment:
> Having a single DebugConfigurationProvider would be cleaner, for sure. However we need the vscode-jest-tests provider such that the Debugconfigurationprovider appears as a separate entry when initially creating a launch.json config and selecting the environment.

you can accomplish this via registering the same DebugConfigurationProvider under both "node" and "vscode-jest-tests" (see comments in `extension.ts`), without splitting the DebugConfigurationProvider...